framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -593,195 +484,306 @@ public void standardizeVariable(AnnotatedTypeMirror type, Element elt) {
                     return;
                 }
 
-                standardizeVariable(declarationTree, type, elt);
+                atVariableDeclaration(declarationTree, type, elt);
                 return;
 
             default:
-                // It's not a variable (it might be METHOD, CONSTRUCTOR, CLASS, or INTERFACE, for
+                // It's not a local variable (it might be METHOD, CONSTRUCTOR, CLASS, or INTERFACE,
+                // for
                 // example), so there is nothing to do.
                 break;
         }
     }
 
     /**
-     * Parse and standardize the expressions in dependent types in {@code type} as if they were
-     * written at {@code localVarPath}. For example, {@code @KeyFor("field") String} is changed to
-     * {@code @KeyFor("this.field")}
+     * Viewpoint-adapt a type qualifier obtained from a contract using {@code stringToJavaExpr}.
      *
-     * @param localVarPath the expression is parsed as if it were written at this location
-     * @param type the type to parse; is side-effected by this method
+     * <p>For example, if the contract is {@code @EnsuresKeyFor(value = "this.field", map =
+     * "this.map")}, this method applies {@code stringToJava} to "this.map" and returns a new {@code
+     * KeyFor} annotation with the result.
+     *
+     * @param annoFromContract an annotation from a contract
+     * @param stringToJavaExpr function used to convert strings to {@link JavaExpression}s
+     * @param errorTree if non-null, where to report any errors that occur when parsing the
+     *     dependent type annotation; if null, report no errors
+     * @return the viewpoint-adapted annotation, or the argument if it is not a dependant type
+     *     annotation
      */
-    private void parseToPath(TreePath localVarPath, AnnotatedTypeMirror type) {
-        Tree enclosingClass = TreePathUtil.enclosingClass(localVarPath);
-        TypeMirror enclosingType = TreeUtils.typeOf(enclosingClass);
-
-        JavaExpression receiver = JavaExpression.getPseudoReceiver(localVarPath, enclosingType);
-
-        JavaExpressionContext localContext =
-                new JavaExpressionContext(
-                        receiver,
-                        JavaExpression.getParametersOfEnclosingMethod(localVarPath),
-                        factory.getChecker());
-        standardizeAtm(localContext, localVarPath, type, /*removeErroneousExpressions=*/ false);
+    public AnnotationMirror viewpointAdaptQualifierFromContract(
+            AnnotationMirror annoFromContract,
+            StringToJavaExpression stringToJavaExpr,
+            @Nullable Tree errorTree) {
+        if (!hasDependentAnnotations()) {
+            return annoFromContract;
+        }
+
+        AnnotationMirror standardized = map(stringToJavaExpr, annoFromContract);
```

Comment:
Methods `map` and `convertAnnotatedTypeMirror` are very similar.  It might be nice for their documentation to mention one another to clarify the relationship, and it might be nice for the method names to be more similar given the method similarities.